dropwizard-db/src/main/java/io/dropwizard/db/AbstractDataSourceFactory.java

```diff
@@ -0,0 +1,66 @@
+package io.dropwizard.db;
+
+import com.codahale.metrics.MetricRegistry;
+import com.google.common.base.Optional;
+import io.dropwizard.util.Duration;
+
+import java.util.Map;
+
+/**
+ * Interface of an abstract JDBC connection pool
+ */
+public interface AbstractDataSourceFactory {
```

Comment:
Sure. I didn't have anything in mind which ends in 'able', so decided to go with 'Abstract'.
`PooledDataSourceFactory` also looks good to me. Naming things is hard...